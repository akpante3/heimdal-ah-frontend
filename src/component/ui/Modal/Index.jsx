import React, { Fragment, Component } from 'react';
import { Modal, ModalHeader, ModalBody } from 'reactstrap';
import './TriggerModal.scss';
import Button from '../Buttons/Button';

/**
 * @description - Heimdal Authors TriggerModal Component
 * @param {props} type - the class of the button(bootstrap or custom)
 * @param {props} label - the label of the modal button
 * @param {props} className - the class of the modal
 * @param {props} children - the children components to be passed
 * @param {function} toggle - the function to be performed when the modal is triggered
 * @returns {fragment} Modal fragment
 */

class TriggerModal extends Component {
  state = {
    display: false,
  };

  toggle = () => {
    this.setState(prevState => ({
      display: !prevState.display,
    }));
  };

  render() {
    return (
      <Fragment>
        <Button
          className={this.props.btnClass}
          type={this.props.type}
          label={this.props.label}
          onClick={this.toggle}
        />
        <Modal
          isOpen={this.state.display}
          toggle={this.toggle}
          className={this.props.className}
        >
          <ModalHeader toggle={this.toggle} />
          <ModalBody>{this.props.children}</ModalBody>
        </Modal>
      </Fragment>
    );
  }
}

export default TriggerModal;
