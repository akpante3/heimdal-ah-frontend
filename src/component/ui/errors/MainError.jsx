import React from 'react';

const MainError = ({ errors }) => (
  <div className="col-md-12 text-danger text-center">
    {errors && errors.mainError && <span className="">{errors.mainError}</span>}
  </div>
);

export default MainError;
