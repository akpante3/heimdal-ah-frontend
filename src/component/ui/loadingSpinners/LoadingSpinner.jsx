import React from 'react';
import { FadeLoader } from 'react-spinners';
import './LoadingSpinner.scss';

const override = `
  display: block;
  margin: 0 auto;
`;

const LoadingSpinner = ({ isLoading, size, color }) => (
  <div className="sweet-loading">
    <FadeLoader
      css={override}
      sizeUnit="px"
      size={size || 100}
      color={color || '#2C4D45'}
      loading={isLoading}
    />
  </div>
);

export default LoadingSpinner;
