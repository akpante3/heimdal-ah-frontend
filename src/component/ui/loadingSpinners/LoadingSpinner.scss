.sweet-loading {
  margin-left: 0px !important;
  margin-top: 0px !important;
}
