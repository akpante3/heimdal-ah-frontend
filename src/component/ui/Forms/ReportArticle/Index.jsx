/* eslint-disable react/forbid-prop-types */
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { setErrors, removeAnError, clearErrors } from '../../../../actions/errorActions';
import { toggleLoader } from '../../../../actions/loaderActions';
import LoadingSpinner from '../../loadingSpinners/LoadingSpinner';
import MainError from '../../errors/MainError';
import { validateReport } from '../../../../helpers/validateInputs';
import { reportArticle } from '../../../../actions/articleActions';
import ToasterAlert from '../../Alert/Alert';

/**
 * @description It wraps its children with a layout style
 * @param {object} props - react props with children
 * @returns {object} the SignupFrom component
 */
class ReportArticle extends Component {
  state = {
    context: '',
    reportType: 'spam',
    errors: {},
    showAlert: false,
  };

  componentWillReceiveProps(nextProps) {
    return nextProps.errors && this.setState({ errors: nextProps.errors });
  }

  onChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });

    if (this.state.errors[event.target.name]) {
      const { actions } = this.props;
      actions.removeAnError(event.target.name);
    }
  };

  flashMessage = () => {
    this.setState({ showAlert: true });

    setTimeout(() => {
      this.setState({ showAlert: false });
    }, 100);
  };

  handleArticleReport = (event) => {
    event.preventDefault();
    const { actions, toggle } = this.props;
    actions.toggleLoader();

    const { context, reportType } = this.state;
    const reportData = { context, reportType };
    const errors = validateReport(reportData);

    if (errors) {
      actions.toggleLoader();
      return actions.setErrors(errors);
    }

    reportData.articleId = this.props.articleId || 5;
    actions.clearErrors();
    return actions.reportArticle(reportData, toggle, this.flashMessage);
  };

  render() {
    const { errors = {}, isLoading } = this.props;

    return (
      <div className="heimdal-report-form">
        {this.state.showAlert && (
          <ToasterAlert
            type="success"
            title="Article Reported"
            message="Your report has been sent successfully"
          />
        )}
        <h1 className="form-title font-cardo text-center">Report this article</h1>
        <form className="heimdal-form" onSubmit={this.handleArticleReport} noValidate>
          <div className="font-cardo ph-30">
            <div className="form-group">
              <select
                defaultValue="spam"
                onChange={this.onChange}
                name="reportType"
                className="form-control"
              >
                <option>spam</option>
                <option>plagarism</option>
                <option>others</option>
              </select>

              {errors.reportType && errors.reportType.constructor === Object && (
                <small className="text-danger h5">{errors.reportType.msg}</small>
              )}
              {errors.reportType && errors.reportType.constructor === Array && (
                <small className="text-danger h5">{errors.reportType}</small>
              )}
            </div>
            <div className="form-group">
              <textarea
                className="form-control"
                name="context"
                onChange={this.onChange}
                placeholder="Type your report message"
              />
              {errors.context && errors.context.constructor === Object && (
                <small className="text-danger h5">{errors.context.msg}</small>
              )}
              {errors.context && errors.context.constructor === Array && (
                <small className="text-danger h5">{errors.context}</small>
              )}
            </div>

            <div className="row">
              <LoadingSpinner isLoading={isLoading} />
              <MainError errors={errors} />
              <div className="col-md-12 text-right">
                <button type="submit" className="btn btn-danger ph-25">
                  Submit Report
                </button>
              </div>
            </div>
          </div>
        </form>
      </div>
    );
  }
}

ReportArticle.defaultProps = {
  errors: {},
  isLoading: false,
};

ReportArticle.propTypes = {
  articleId: PropTypes.number.isRequired,
  isLoading: PropTypes.bool,
  errors: PropTypes.object,
  toggle: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  errors: state.errors,
  isLoading: state.loader.isLoading,
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    {
      removeAnError,
      setErrors,
      clearErrors,
      reportArticle,
      toggleLoader,
    },
    dispatch,
  ),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ReportArticle);
