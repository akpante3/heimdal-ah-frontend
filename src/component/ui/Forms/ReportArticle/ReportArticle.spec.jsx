/* eslint-disable no-undef */
import React from 'react';
// import sinon from 'sinon';
import { ReportArticle } from './Index';
import LoadingSpinner from '../../loadingSpinners/LoadingSpinner';
import MainError from '../../errors/MainError';

describe('The ReportArticle Component Test Suite', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<ReportArticle toggle={() => {}} articleId={5} actions={() => {}} />);
  });
  it('should render the SignupFrom component', () => {
    expect(wrapper.length).toBe(1);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(ReportArticle)).toBeDefined();
  });

  it('should render exactly one form tag', () => {
    expect(wrapper.find('form')).toHaveLength(1);
  });

  it('it should render exactly one <LoadingSpinner /> component', () => {
    expect(wrapper.find(LoadingSpinner)).toHaveLength(1);
  });

  it('it should render exactly one <MainError /> component', () => {
    expect(wrapper.find(MainError)).toHaveLength(1);
  });

  //= ========

  it('should handle input change event for context', () => {
    const event = {
      target: { name: 'context', value: 'henry is a very good boy' },
      preventDefault: () => {},
    };

    wrapper.instance().onChange(event);
    expect(wrapper.state().context).toEqual('henry is a very good boy');
    expect(wrapper.instance().state.context).toEqual(event.target.value);
  });

  it('should handle input change event for reportType', () => {
    const event = { target: { name: 'reportType', value: 'spam' }, preventDefault: () => {} };

    wrapper.instance().onChange(event);
    expect(wrapper.state().reportType).toEqual('spam');
    expect(wrapper.instance().state.reportType).toEqual(event.target.value);
  });

  it('should call the handleArticleReport', () => {
    const event = { preventDefault: () => {} };

    wrapper.instance().handleArticleReport(event);
    expect(wrapper.state().reportType).toEqual('spam');
  });
});
