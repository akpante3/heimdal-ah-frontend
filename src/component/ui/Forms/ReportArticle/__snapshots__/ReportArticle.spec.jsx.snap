// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ReportArticle Component Test Suite should render the SignupFrom component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReportArticle
    actions={[Function]}
    articleId={5}
    errors={Object {}}
    isLoading={false}
    toggle={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <h1
          className="form-title font-cardo text-center"
        >
          Report this article
        </h1>,
        <form
          className="heimdal-form"
          noValidate={true}
          onSubmit={[Function]}
        >
          <div
            className="font-cardo ph-30"
          >
            <div
              className="form-group"
            >
              <select
                className="form-control"
                defaultValue="spam"
                name="reportType"
                onChange={[Function]}
              >
                <option>
                  spam
                </option>
                <option>
                  plagarism
                </option>
                <option>
                  others
                </option>
              </select>
            </div>
            <div
              className="form-group"
            >
              <textarea
                className="form-control"
                name="context"
                onChange={[Function]}
                placeholder="Type your report message"
              />
            </div>
            <div
              className="row"
            >
              <LoadingSpinner
                isLoading={false}
              />
              <MainError
                errors={Object {}}
              />
              <div
                className="col-md-12 text-right"
              >
                <button
                  className="btn btn-danger ph-25"
                  type="submit"
                >
                  Submit Report
                </button>
              </div>
            </div>
          </div>
        </form>,
      ],
      "className": "heimdal-report-form",
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Report this article",
          "className": "form-title font-cardo text-center",
        },
        "ref": null,
        "rendered": "Report this article",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="font-cardo ph-30"
          >
            <div
              className="form-group"
            >
              <select
                className="form-control"
                defaultValue="spam"
                name="reportType"
                onChange={[Function]}
              >
                <option>
                  spam
                </option>
                <option>
                  plagarism
                </option>
                <option>
                  others
                </option>
              </select>
            </div>
            <div
              className="form-group"
            >
              <textarea
                className="form-control"
                name="context"
                onChange={[Function]}
                placeholder="Type your report message"
              />
            </div>
            <div
              className="row"
            >
              <LoadingSpinner
                isLoading={false}
              />
              <MainError
                errors={Object {}}
              />
              <div
                className="col-md-12 text-right"
              >
                <button
                  className="btn btn-danger ph-25"
                  type="submit"
                >
                  Submit Report
                </button>
              </div>
            </div>
          </div>,
          "className": "heimdal-form",
          "noValidate": true,
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="form-group"
              >
                <select
                  className="form-control"
                  defaultValue="spam"
                  name="reportType"
                  onChange={[Function]}
                >
                  <option>
                    spam
                  </option>
                  <option>
                    plagarism
                  </option>
                  <option>
                    others
                  </option>
                </select>
              </div>,
              <div
                className="form-group"
              >
                <textarea
                  className="form-control"
                  name="context"
                  onChange={[Function]}
                  placeholder="Type your report message"
                />
              </div>,
              <div
                className="row"
              >
                <LoadingSpinner
                  isLoading={false}
                />
                <MainError
                  errors={Object {}}
                />
                <div
                  className="col-md-12 text-right"
                >
                  <button
                    className="btn btn-danger ph-25"
                    type="submit"
                  >
                    Submit Report
                  </button>
                </div>
              </div>,
            ],
            "className": "font-cardo ph-30",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <select
                    className="form-control"
                    defaultValue="spam"
                    name="reportType"
                    onChange={[Function]}
                  >
                    <option>
                      spam
                    </option>
                    <option>
                      plagarism
                    </option>
                    <option>
                      others
                    </option>
                  </select>,
                  undefined,
                  undefined,
                ],
                "className": "form-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <option>
                        spam
                      </option>,
                      <option>
                        plagarism
                      </option>,
                      <option>
                        others
                      </option>,
                    ],
                    "className": "form-control",
                    "defaultValue": "spam",
                    "name": "reportType",
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "spam",
                      },
                      "ref": null,
                      "rendered": "spam",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "plagarism",
                      },
                      "ref": null,
                      "rendered": "plagarism",
                      "type": "option",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "others",
                      },
                      "ref": null,
                      "rendered": "others",
                      "type": "option",
                    },
                  ],
                  "type": "select",
                },
                undefined,
                undefined,
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <textarea
                    className="form-control"
                    name="context"
                    onChange={[Function]}
                    placeholder="Type your report message"
                  />,
                  undefined,
                  undefined,
                ],
                "className": "form-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-control",
                    "name": "context",
                    "onChange": [Function],
                    "placeholder": "Type your report message",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "textarea",
                },
                undefined,
                undefined,
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <LoadingSpinner
                    isLoading={false}
                  />,
                  <MainError
                    errors={Object {}}
                  />,
                  <div
                    className="col-md-12 text-right"
                  >
                    <button
                      className="btn btn-danger ph-25"
                      type="submit"
                    >
                      Submit Report
                    </button>
                  </div>,
                ],
                "className": "row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "isLoading": false,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "errors": Object {},
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <button
                      className="btn btn-danger ph-25"
                      type="submit"
                    >
                      Submit Report
                    </button>,
                    "className": "col-md-12 text-right",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Submit Report",
                      "className": "btn btn-danger ph-25",
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": "Submit Report",
                    "type": "button",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <h1
            className="form-title font-cardo text-center"
          >
            Report this article
          </h1>,
          <form
            className="heimdal-form"
            noValidate={true}
            onSubmit={[Function]}
          >
            <div
              className="font-cardo ph-30"
            >
              <div
                className="form-group"
              >
                <select
                  className="form-control"
                  defaultValue="spam"
                  name="reportType"
                  onChange={[Function]}
                >
                  <option>
                    spam
                  </option>
                  <option>
                    plagarism
                  </option>
                  <option>
                    others
                  </option>
                </select>
              </div>
              <div
                className="form-group"
              >
                <textarea
                  className="form-control"
                  name="context"
                  onChange={[Function]}
                  placeholder="Type your report message"
                />
              </div>
              <div
                className="row"
              >
                <LoadingSpinner
                  isLoading={false}
                />
                <MainError
                  errors={Object {}}
                />
                <div
                  className="col-md-12 text-right"
                >
                  <button
                    className="btn btn-danger ph-25"
                    type="submit"
                  >
                    Submit Report
                  </button>
                </div>
              </div>
            </div>
          </form>,
        ],
        "className": "heimdal-report-form",
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Report this article",
            "className": "form-title font-cardo text-center",
          },
          "ref": null,
          "rendered": "Report this article",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="font-cardo ph-30"
            >
              <div
                className="form-group"
              >
                <select
                  className="form-control"
                  defaultValue="spam"
                  name="reportType"
                  onChange={[Function]}
                >
                  <option>
                    spam
                  </option>
                  <option>
                    plagarism
                  </option>
                  <option>
                    others
                  </option>
                </select>
              </div>
              <div
                className="form-group"
              >
                <textarea
                  className="form-control"
                  name="context"
                  onChange={[Function]}
                  placeholder="Type your report message"
                />
              </div>
              <div
                className="row"
              >
                <LoadingSpinner
                  isLoading={false}
                />
                <MainError
                  errors={Object {}}
                />
                <div
                  className="col-md-12 text-right"
                >
                  <button
                    className="btn btn-danger ph-25"
                    type="submit"
                  >
                    Submit Report
                  </button>
                </div>
              </div>
            </div>,
            "className": "heimdal-form",
            "noValidate": true,
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="form-group"
                >
                  <select
                    className="form-control"
                    defaultValue="spam"
                    name="reportType"
                    onChange={[Function]}
                  >
                    <option>
                      spam
                    </option>
                    <option>
                      plagarism
                    </option>
                    <option>
                      others
                    </option>
                  </select>
                </div>,
                <div
                  className="form-group"
                >
                  <textarea
                    className="form-control"
                    name="context"
                    onChange={[Function]}
                    placeholder="Type your report message"
                  />
                </div>,
                <div
                  className="row"
                >
                  <LoadingSpinner
                    isLoading={false}
                  />
                  <MainError
                    errors={Object {}}
                  />
                  <div
                    className="col-md-12 text-right"
                  >
                    <button
                      className="btn btn-danger ph-25"
                      type="submit"
                    >
                      Submit Report
                    </button>
                  </div>
                </div>,
              ],
              "className": "font-cardo ph-30",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <select
                      className="form-control"
                      defaultValue="spam"
                      name="reportType"
                      onChange={[Function]}
                    >
                      <option>
                        spam
                      </option>
                      <option>
                        plagarism
                      </option>
                      <option>
                        others
                      </option>
                    </select>,
                    undefined,
                    undefined,
                  ],
                  "className": "form-group",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <option>
                          spam
                        </option>,
                        <option>
                          plagarism
                        </option>,
                        <option>
                          others
                        </option>,
                      ],
                      "className": "form-control",
                      "defaultValue": "spam",
                      "name": "reportType",
                      "onChange": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "spam",
                        },
                        "ref": null,
                        "rendered": "spam",
                        "type": "option",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "plagarism",
                        },
                        "ref": null,
                        "rendered": "plagarism",
                        "type": "option",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "others",
                        },
                        "ref": null,
                        "rendered": "others",
                        "type": "option",
                      },
                    ],
                    "type": "select",
                  },
                  undefined,
                  undefined,
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <textarea
                      className="form-control"
                      name="context"
                      onChange={[Function]}
                      placeholder="Type your report message"
                    />,
                    undefined,
                    undefined,
                  ],
                  "className": "form-group",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control",
                      "name": "context",
                      "onChange": [Function],
                      "placeholder": "Type your report message",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "textarea",
                  },
                  undefined,
                  undefined,
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <LoadingSpinner
                      isLoading={false}
                    />,
                    <MainError
                      errors={Object {}}
                    />,
                    <div
                      className="col-md-12 text-right"
                    >
                      <button
                        className="btn btn-danger ph-25"
                        type="submit"
                      >
                        Submit Report
                      </button>
                    </div>,
                  ],
                  "className": "row",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "isLoading": false,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "errors": Object {},
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <button
                        className="btn btn-danger ph-25"
                        type="submit"
                      >
                        Submit Report
                      </button>,
                      "className": "col-md-12 text-right",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Submit Report",
                        "className": "btn btn-danger ph-25",
                        "type": "submit",
                      },
                      "ref": null,
                      "rendered": "Submit Report",
                      "type": "button",
                    },
                    "type": "div",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
